webapp/src/ts/services/form.service.ts

```diff
@@ -335,6 +372,16 @@ export class FormService {
     const preparedDocs = await this.applyTransitions(docs);
 
     const primaryDoc = preparedDocs.preparedDocs.find(doc => doc.type === type);
+
+    const duplicates = await this.checkForDuplicates(
+      primaryDoc || preparedDocs.preparedDocs[0], 
+      duplicateCheck, 
+      acknowledged
+    );
+    if (duplicates && duplicates.length > 0){
```

Comment:
nitpick: if we keep the return type for `checkForDuplicates` consistent, then we can simplify this to just:

```suggestion
    if (duplicates.length){
```